src/server/db_slice.cc

```diff
@@ -660,6 +660,19 @@ void DbSlice::PostUpdate(DbIndex db_ind, PrimeIterator it, bool existing) {
     stats->strval_memory_usage += value_heap_size;
   if (existing)
     stats->update_value_amount += value_heap_size;
+
+
+  // Check if the key is watched.
+  // TODO: eliminate ToString() call
+  auto key = it->first.ToString();
+  VLOG(0) << "post update for key: " << key;
+  if(auto wit = watched_keys_.find(key); wit != watched_keys_.end()) {
+    VLOG(0) << "watched key touched " << key << " dependants " << wit->second.size();
+    for (auto conn_ptr: wit->second) conn_ptr->watched_dirty.store(true);
+    // TODO: watched_keys is not updated accordingly due to being shared accross threads - just ignore?
+    // No connections need to watch it anymore.
+    watched_keys_.erase(wit);
+  }
```

Comment:
See my TODO (by watched_keys I mean conn_ptr->watched_keys)

The connection will loop over those values on cleanup even if they were evicted. But any solution I can think of is more costly.

Reply:
it's totally fine.